delta-lake/src/main/java/org/apache/iceberg/delta/BaseMigrateDeltaLakeTableAction.java

```diff
@@ -0,0 +1,356 @@
+    Map<String, List<Action>> deltaLakeActionMap =
+        actions.stream()
+            .filter(action -> action instanceof AddFile || action instanceof RemoveFile)
+            .collect(Collectors.groupingBy(a -> a.getClass().getSimpleName()));
+
+    List<DataFile> filesToAdd = Lists.newArrayList();
+    List<DataFile> filesToRemove = Lists.newArrayList();
+    for (Action action : Iterables.concat(deltaLakeActionMap.values())) {
+      DataFile dataFile = buildDataFileFromAction(action, transaction.table());
+      if (action instanceof AddFile) {
+        filesToAdd.add(dataFile);
+      } else if (action instanceof RemoveFile) {
+        filesToRemove.add(dataFile);
+      } else {
+        throw new ValidationException(
+            "The action %s's is unsupported", action.getClass().getSimpleName());
+      }
+    }
+
+    if (filesToAdd.size() > 0 && filesToRemove.size() > 0) {
+      // Overwrite_Files case
+      OverwriteFiles overwriteFiles = transaction.newOverwrite();
+      filesToAdd.forEach(overwriteFiles::addFile);
+      filesToRemove.forEach(overwriteFiles::deleteFile);
+      overwriteFiles.commit();
+    } else if (filesToAdd.size() > 0) {
+      // Append_Files case
+      AppendFiles appendFiles = transaction.newAppend();
+      filesToAdd.forEach(appendFiles::appendFile);
+      appendFiles.commit();
+    } else if (filesToRemove.size() > 0) {
+      // Delete_Files case
+      DeleteFiles deleteFiles = transaction.newDelete();
+      filesToRemove.forEach(deleteFiles::deleteFile);
+      deleteFiles.commit();
+    }
+  }
+
+  public DataFile buildDataFileFromAction(Action action, Table table) {
+    PartitionSpec spec = table.spec();
+    String path;
+    Map<String, String> partitionValues;
+
+    if (action instanceof AddFile) {
+      AddFile addFile = (AddFile) action;
+      path = addFile.getPath();
+      partitionValues = addFile.getPartitionValues();
+    } else if (action instanceof RemoveFile) {
+      RemoveFile removeFile = (RemoveFile) action;
+      path = removeFile.getPath();
+      partitionValues = removeFile.getPartitionValues();
+    } else {
+      throw new IllegalStateException(
+          String.format(
+              "Unexpected action type for Delta Lake: %s", action.getClass().getSimpleName()));
+    }
+
+    String fullFilePath = getFullFilePath(path, deltaLog.getPath().toString());
+
+    if (partitionValues == null) {
+      // For unpartitioned table, the partitionValues should be an empty map rather than null
+      throw new ValidationException("File %s does not specify a partitionValues", fullFilePath);
+    }
+
+    FileFormat format = determineFileFormatFromPath(fullFilePath);
+    FileIO io = table.io();
+    InputFile file;
+    if (io != null) {
+      file = io.newInputFile(fullFilePath);
+    } else {
+      file = HadoopInputFile.fromPath(new Path(fullFilePath), this.hadoopConfiguration);
+    }
+
+    MetricsConfig metricsConfig = MetricsConfig.forTable(table);
```

Comment:
these values are always null since it's a new table. I guess for now we can just use default metrics config and null name mapping. If there is a need we can make that user configurable later.

Reply:
I think they may be already user configurable. Our implementation currently have `tableProperty` and `tableProperties` to allow users to add table properties before executing the migration. For example, user can provide their namemapping string through 
`migrationaction.tableProperty("schema.name-mapping.default", <json-string>).execute()`